src/main/java/de/tum/in/www1/artemis/domain/participation/Participation.java

```diff
@@ -279,7 +279,7 @@ private <T extends Submission> Optional<T> findLatestSubmission(boolean includeI
         }
 
         return (Optional<T>) submissions.stream().max((s1, s2) -> {
-            if (s1.getSubmissionDate() == null || s2.getSubmissionDate() == null) {
+            if (s1.getSubmissionDate() == null || s2.getSubmissionDate() == null || s1.getSubmissionDate().equals(s2.getSubmissionDate())) {
```

Comment:
This does not resolve the issue for me. I guess thats the case because the submission dates are not equal, but hashsums are.

```
mysql> select * from submission where id=656 or id=657;
+---------------+-----+----------+----------------------+-----------------+------------------------------------------+------------------+----------------------------+-----------+------+------------------+----------------------------------------+-------+----------+----------------------------+--------------------------------+------------+
| discriminator | id  | jhi_type | submitted            | score_in_points | commit_hash                              | participation_id | submission_date            | file_path | text | explanation_text | example_submission                     | model | language | build_failed               | build_artifact                 | quiz_batch |
+---------------+-----+----------+----------------------+-----------------+------------------------------------------+------------------+----------------------------+-----------+------+------------------+----------------------------------------+-------+----------+----------------------------+--------------------------------+------------+
| P             | 656 | MANUAL   | 0x01                 |            NULL | 9bca4d27a373bb47fa7fd4cbb3a22cfa61156049 |              285 | 2022-07-19 11:34:56.017944 | NULL      | NULL | NULL             | NULL                                   | NULL  | NULL     | 0x00                       | 0x00                           |       NULL |
| P             | 657 | MANUAL   | 0x01                 |            NULL | 9bca4d27a373bb47fa7fd4cbb3a22cfa61156049 |              285 | 2022-07-19 11:34:56.044334 | NULL      | NULL | NULL             | NULL                                   | NULL  | NULL     | 0x00                       | 0x00                           |       NULL |
+---------------+-----+----------+----------------------+-----------------+------------------------------------------+------------------+----------------------------+-----------+------+------------------+----------------------------------------+-------+----------+----------------------------+--------------------------------+------------+
```